docs/src/main/paradox/docs/delta/architecture.md

```diff
@@ -1,14 +1,150 @@
 # Architecture
 
-## Service anatomy
+Blue Brain Nexus is a collection of software components that address various organizational needs relating to data
+storage, management, analysis and consumption. It was designed to support the data-driven science iterative cycle at
+Blue Brain but its genericity allows for its use in arbitrary contexts.
 
+This document focuses on the characteristics of the Nexus Delta and its design choices.
 
-Nexus Delta is assembled by a series of components
+## Components
 
-- **Write Model**: This component is executed when a customer performs a write API operation (create, update, tag, ...). The business logic is triggered (validation, state computation, Json and Graph manipulations, etc). As a result, an Event gets stored into the Primary Store or a rejection is delivered to the client.
-- **Event Stream**: This is an asynchronous process that gets triggered as soon as there are elements to be read from the Primary Store. The stream runs several operations that take the stored Events and transform them. The result is then saved into a target index.
-- **Read Model**: This component gets triggered when a customer performs a read API operation (get, list). A call to the index is performed. The resulting data is transformed if required and delivered to the client.
+Nexus Delta is a low latency, scalable and secure service that realizes a range of functions to support data management
+and knowledge graph lifecycles. It is a central piece of the Blue Brain Nexus ecosystem of software components as it
```

Comment:
This block is kind of duplicated from the index.md.

I would shorten it to just:

```
Nexus Delta is a low latency, scalable and secure service that realizes a range of functions to support data management and knowledge graph lifecycles. On order to achieve its goals it uses [Apache Cassandra] as a primary store
(source of truth for all the information in the system), [ElasticSearch] for full text search and [BlazeGraph] for graph
based data access.
```